lib/clearance/configuration.rb

```diff
@@ -44,23 +121,36 @@ def  user_actions
       end
     end
 
+    # The name of foreign key parameter for the configured user model.
+    # This is derived from the `model_name` of the `user_model` setting.
+    # In the default configuration, this is `user_id`.
+    # @return [Symbol]
     def user_id_parameter
       "#{user_model.model_name.singular}_id".to_sym
     end
 
+    # @return [Boolean] are Clearance's built-in routes enabled?
     def routes_enabled?
       @routes
     end
   end
 
+  # @return [Clearance::Configuration] Clearance's current configuration
   def self.configuration
     @configuration ||= Configuration.new
   end
 
+  # Set Clearance's configuration
+  # @param config [Clearance::Configuration]
   def self.configuration=(config)
     @configuration = config
   end
 
+  # Modify Clearance's current configuration
+  #   Clearance.configure do |config|
```

Comment:
This code sample might not render correctly, you might have to indent this by four spaces to do so.